### PR TITLE
Increase PSK_MAX_IDENTITY_LEN from 128 to 256

### PR DESCRIPTION
**Pull request by ericcurtin**
We are considering using the format "host-nqn controller-nqn" for
psk-id in the NVMe-oF/TCP over TLS spec, it's in the current version,
but openssl's limit was 128 upto now, we need a little longer than that.

### CODE REVIEW

**Comment by ericcurtin** [comment]
Coincidentally someone tried to increase PSK_MAX_PSK_LEN in #12758 recently

What rfc4279 says about this:

```
5.3.  Requirements for TLS Implementations

   TLS implementations supporting these ciphersuites MUST support
   arbitrary PSK identities up to 128 octets in length, and arbitrary
   PSKs up to 64 octets in length.  Supporting longer identities and
   keys is RECOMMENDED
```

Funnily enough the rfc recommends the psk-id should be double the length of the psk (while in openssl it's half the length of the psk), and recommends supporting longer, so I could change it to 512 if possible also?

**Comment by richsalz** [comment]
I would ask what other TLS implementations do, so that you don't run into interop issues.

**Comment by ericcurtin** [comment]
Yeah, I've looked into that, most support 127 or 128 excluding null terminator

**Comment by richsalz** [comment]
Since most, including OpenSSL, support 12[78] then shouldn't you live within that limit?  For interop.

**Comment by ericcurtin** [comment]
We are considering living with that limit but after seeing #12758 also, I'm starting to suspect some of these limits may be starting to become long in the tooth, not just for my use case.

**Comment by ericcurtin** [comment]
Hi @slontis @richsalz I have no idea how the otc review process works. When will this go through otc review?

**Comment by slontis** [comment]
@mattcaswell - are you happy with this change?
I don't know if there are any side effects because of this.

**Comment by ericcurtin** [comment]
Hi @mattcaswell @slontis and @richsalz 

I see ssl.h got renamed to ssl.h.in. I can fix the conflicts no biggie :smile: But would it be possible to get this through otc review stage?

**Review by mattcaswell — APPROVED** [review]
I'm ok with this. Approved - needs second review.

**Review by slontis — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed. I resolved a trivial merge conflict while doing so.

**Comment by ericcurtin** [comment]
Thanks for getting around to this @mattcaswell

**Pull request closed without merging**
